Update libmesh

lindsayad opened the pull request:
Get double curly braces for std::array initialization so we can pass our
minimu clang requirements.

lindsayad committed:
Update libmesh

Get double curly braces for std::array initialization so we can pass our
minimu clang requirements.

Refs #000

dschwen commented:
@lindsayad I think this will break die to the deprecated numerical `size()` method on vector values` being removed

lindsayad commented:
K I'll work on getting things updated. Any broken apps can have their use
of "size" replaced with "norm" ahead of this PR getting merged

On Mon, Jun 1, 2020 at 4:30 PM Daniel Schwen <notifications@github.com>
wrote:

> @lindsayad <https://github.com/lindsayad> I think this will break die to
> the deprecated numerical size() method on vector values` being removed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/15373#issuecomment-637179384>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACOGA4ETH45OHGSBYMQUXZ3RUQ2X7ANCNFSM4NQGMMJA>
> .
>

lindsayad committed:
TypeVector::size -> TypeVector::norm

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/537534/) on 03650cc wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/15373/site)

This comment will be updated on new commits.

dschwen commented:
I have a Marmot fix ready

lindsayad committed:
Update meta.yaml

dschwen commented:
Something is screwy with the Magpie recipe... @milljm 
but it's not related.

https://civet.inl.gov/job/538224/

dschwen commented on conda/libmesh/meta.yaml:
```diff
@@ -1,6 +1,6 @@
 {% set build = 0 %}
 {% set strbuild = "build_" + build|string %}
-{% set version = "2020.05.21" %}
+{% set version = "2020.06.2" %}
```
should this be `2020.06.02` ?

dschwen reviewed (dismissed):
If the package version is ok with a single digit day this is good to go.

lindsayad commented:
I'll make it `2020.06.02`

lindsayad committed:
Strip debug symbols from conda-generated libmesh libraries

These symbols take up disk space and they are immediately garbage once
the package is distributed without the source code

Closes #15327

lindsayad commented:
Hahaha you keep approving it but I keep negating it. I had the wrong LDFLAGS for stripping the symbols before. Should be good now.

@loganharbour hopefully this build looks to your satisfaction?

loganharbour reviewed (approved):
👍  let's hope this is finally done!

We should probably start preparing some documentation about use of `moose-libmesh` for running vs development

dschwen commented:
Uh, this builds all methods now and then strips symbols? I'm sure there is some wisdom in that... debugging just moose without symbol errors?

loganharbour commented:
Without libMesh symbol errors, which were already not usable.
